modules/exploits/linux/http/fusionpbx_php_editor_cmd_exec.rb

```diff
@@ -0,0 +1,135 @@
+####################################################################
+# This module requires Metasploit: https://metasploit.com/download #
+#  Current source: https://github.com/rapid7/metasploit-framework  #
+####################################################################
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+        "Name" => "FusionPBX =< 4.5.10 Authenticated Remote Code Execution",
+        "Description" =>  %q{
+          Authenticated Remote Code Execution in FusionPBX web appliances.
+          Affected versions: =< 4.5.10
+          By using the on site PHP-Editor function users can overwrite PHP files on the web server
+          with a trojan horse and get a shell as www-data by navigating to the changed PHP file location.
+          Vendor verified: 11/27/2019
+          Vendor patched: 12/09/2019
+          Public disclosure: 01/20/2020
+        },
+        "License" => MSF_LICENSE,
+        'Author' => [
+          'enjloezz', # Metasploit Module
+          'TheCyberGeek' # Discovery and Metasploit Module
+        ],
+        'References' =>
+        [
+            ['URL','https://github.com/fusionpbx/fusionpbx/releases'],
+            ['CVE','2019-19369']
+        ],
+        "Platform"	=> "php",
+        "Arch"		=> ARCH_PHP,
+        "Targets" => [
+          ["FusionPBX", {}],
+        ],
+        "Stance" => Msf::Exploit::Stance::Aggressive,
+        "Privileged" => false,
+        "DisclosureDate" => "Jan 21 2020",
+        "DefaultTarget" => 0
+      ))
+    register_options(
+      [
+        OptString.new("TARGETURI", [true, "The URI of the FusionPBX Application", "/"]),
+        OptString.new("USERNAME", [true, "The Username of the FusionPBX Application", "admin"]),
+        OptString.new("PASSWORD", [true, "The Password of the FusionPBX Application", "fusion"]),
+      ]
+    )
+  end
+
```

Comment:
Is it possible to include a `check` command?  Is the version number of the application disclosed anywhere that you could pull?  Maybe even just checking the login page says 'fusionpbx' would suffice at a VERY minimum?

Reply:
There is a system config page in settings that displays version numbers. I will add the check shortly.

Reply:
Check added

Reply:
I thought it would be appropriate to check if the PHP editor exists rather then the version, I originally discovered this on version 4.4.5.